src/Policeman/Core/Package.hs

```diff
@@ -16,3 +23,28 @@ newtype PackageName = PackageName
 newtype Version = Version
     { unVersion :: Text
     }
+
+{- | Overall structure of the package.
+Generally speaking, this is the resulting data type of custom parsing the
+project by @policeman@.
+-}
+data PackageStructure = PackageStructure
+    { psModules    :: !(Set Module)  -- ^ List of the external modules
+    , psExports    :: !(Set Export)  -- ^ The summary list of the exported stuff.
+    , psModulesMap :: !(HashMap Module ModuleStructure)  -- ^ The pairs of the module and its parsed structure.
+    }
+
+newtype Module = Module
+  { unModule :: Text
+  } deriving newtype (Eq, Ord)
+
+data Export
+    = ExportedFunction Text
+    | ExportedType Text
+    | Exported Module
+    deriving stock (Eq, Ord)
```

Comment:
This looks nice! I generally like the idea of working with our custom data type instead of raw Haskell AST, since we can convert to our convenient form, dropping all irrelevant information, and analyze the only relevant parts